sdk/src/main/scala/com/horizen/account/state/CertificateKeyRotationMsgProcessor.scala

```diff
@@ -147,7 +148,17 @@ case class CertificateKeyRotationMsgProcessor(params: NetworkParams) extends Nat
   private def execSubmitKeyRotation(msg: Message, view: BaseAccountStateView, currentEpochNum: Int): Array[Byte] = {
     //verify
     checkMessageValidity(msg)
-    val inputData = SubmitKeyRotationCmdInputDecoder.decode(getArgumentsFromData(msg.getData))
+
+    val inputData : SubmitKeyRotationCmdInput = Try {
```

Comment:
check the method below `checkKeyRotationProofValidity`.
in case the `keyRotationProof.newKey` is not a valid key then
```
    val newKeyAsMessage = keyRotationProof.keyType match {
      case SigningKeyRotationProofType => CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation
        .getMsgToSignForSigningKeyUpdate(keyRotationProof.newKey.pubKeyBytes(), currentEpochNum, params.sidechainId)
      case MasterKeyRotationProofType => CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation
        .getMsgToSignForMasterKeyUpdate(keyRotationProof.newKey.pubKeyBytes(), currentEpochNum, params.sidechainId)
    }
``` may throw a run time exception.

Let's wrap the method inside with `Try(fn).recoverWith {` and check in case method throws exception not of type `ExecutionFailedException` -> wrap it with `ExecutionRevertedException` and throw outside.

In general we must be careful with sc_cryptolib objects in msg processors.
We may discuss if needed.

Reply:
Done. I also added some more check to `ThresholdSignatureCircuitWithKeyRotationImplZendoo.java` in order to avoid sending bad data to sc crypto lib which might lead to a rust panic event